sklearn/base.py

```diff
@@ -332,10 +336,19 @@ def _get_tags(self):
         return collected_tags
 
 
+def _sizeof(cls):
+    #  return sum(x.nbytes if hasattr(x, 'nbytes') else sys.getsizeof(x)
+    #  for x in cls.__dict__.values())
+    return sum(sys.getsizeof(x) for x in cls.__dict__.values())
+
+
 class ClassifierMixin:
     """Mixin class for all classifiers in scikit-learn."""
     _estimator_type = "classifier"
 
+    def __sizeof__(self):
+        return _sizeof(self)
```

Comment:
Why do we need it in each mixin, wouldn't just `BaseEstimator` be enough? Everything inherits from it.

Reply:
Do all estimators inherit from `BaseEstimator`? It doesn't look like it in the code because `class ClassifierMixin:` is used, not `class ClassifierMixin(BaseEstimator):`. It also doesn't look like it in Python:

``` python
>>> from sklearn.base import ClassifierMixin
>>> ClassifierMixin.mro()
[sklearn.base.ClassifierMixin, object]
```

Reply:
> Do all estimators inherit from BaseEstimator

Estimators - yes. There might be other objects that don't though. E.g. losses (https://github.com/scikit-learn/scikit-learn/blob/557218c021d58c7455b26820f331a9c388c9cb4d/sklearn/_loss/glm_distribution.py#L20)

`ClassifierMixin` is a mixin, so it's never going to be used alone. It's always BaseEstimator + e.g. ClassifierMixin.

So there is no need to add this method to Mixins, you can be sure they are always used in combination with BaseEstimator. However this might need to be done for other custom objects.